src/utils/components/DiskModal/utils/submit.ts

```diff
@@ -0,0 +1,127 @@
+import produce from 'immer';
+
+import { V1beta1DataVolume } from '@kubevirt-ui/kubevirt-api/containerized-data-importer/models';
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { UploadDataProps } from '@kubevirt-utils/hooks/useCDIUpload/useCDIUpload';
+import { getName } from '@kubevirt-utils/resources/shared';
+import {
+  getBootDisk,
+  getDataVolumeTemplates,
+  getDisks,
+  getVolumes,
+} from '@kubevirt-utils/resources/vm';
+import { generatePrettyName, isEmpty } from '@kubevirt-utils/utils/utils';
+import { isRunning } from '@virtualmachines/utils';
+
+import { DEFAULT_DISK_SIZE } from './constants';
+import { getEmptyVMDataVolumeResource, hotplugPromise, produceVMDisks } from './helpers';
+import { V1DiskFormState } from './types';
+
+const makeItBootableVolume = (vm: V1VirtualMachine, diskName: string) => {
+  return produce(vm, (draftVM) => {
+    const disks = getDisks(draftVM);
+
+    disks.find((disk) => disk.name === diskName).bootOrder = 1;
+
+    let bootOrder = 2;
+    disks.forEach((disk) => {
+      if (disk.name !== diskName) {
+        disk.bootOrder = bootOrder;
+        bootOrder += 1;
+      }
+    });
+  });
+};
+
+const noMoreBootableVolume = (vm: V1VirtualMachine, diskName: string) => {
+  return produce(vm, (draftVM) => {
+    const disks = getDisks(draftVM);
+    const disk = disks.find((d) => d.name === diskName);
+
+    if (disk) delete disk.bootOrder;
+
+    const nextBootDisk = disks.find((d) => d.name !== diskName);
+
+    if (nextBootDisk) {
+      nextBootDisk.bootOrder = 1;
+    }
```

Comment:
should we also adjust the other disks or it doesn't matter?

Reply:
does not matter. https://docs.openshift.com/container-platform/4.8/virt/virtual_machines/virt-edit-boot-order.html